artifacts/k8s-templates/konnectivity-proxy.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: konnectivity-proxy
+  namespace: kube-system
+  annotations:
+    scheduler.alpha.kubernetes.io/critical-pod: ''
+    seccomp.security.alpha.kubernetes.io/pod: 'docker/default'
+spec:
+  hostNetwork: true
+  containers:
+  - name: konnectivity-proxy-container
+    image: ${PROXY_IMAGE}:${TAG}
+    resources:
+      requests:
+        cpu: 1m
+    command: [ "/proxy-server"]
+    args: [
+      "--log-file=/var/log/konnectivity-server.log",
+      "--logtostderr=false",
+      "--log-file-max-size=0",
+      "--server-uds=/etc/srv/kubernetes/konnectivity/konnectivity-server.socket",
+      "--cluster-cert=/etc/srv/kubernetes/pki/apiserver.crt",
```

Comment:
Based on the logic in GetTLSConfigwithAPIAuth, the proxy agent is going to use the apiserver's cert as the caCert. Is that the cert the same as this apiserver.crt?

Reply:
As a step #1, agent uses CA certificate provisioned on kublet to communicate with kubeapi-server. As part of tls handshake, kubeapi-server returns a public key (#2) . The "--cluster-cert" on server side is the same private key as used by kubeapi-server, therefore proxy can decrypt messages sent from agent that are encrypted by key returned from #2.